### DISCUSSION TITLE
Adding self for vim.iter?

### DISCUSSION THREAD

**Discussion by kuro337** [neovim/neovim#30702] [Ideas]
I find myself doing this a lot - and I end up using a wrapper inteface around vim.iter to have inline logic - I was curious if this seems like a good idea?

For example:
![image](https://github.com/user-attachments/assets/783b6bb2-741d-471b-a059-accc27393c7d)

As a workaround I end up doing something like this or a wrapper,

![image](https://github.com/user-attachments/assets/93e005f4-68a0-4cba-8454-f8db915463c7)

Also wouldn't this be true to Lua Metatables standard behavior - any object has access to self? 

Or would it cause too many gotcha's with mutation etc? 
Although even currently - within the iteration I noticed calling it:pop() etc. doesn't modify the iteration , so I'm assuming it works on a copy of the list passed during map,etc. instead of a reference?

Would be curious about it!

![image](https://github.com/user-attachments/assets/8dbc97cf-7ea8-4d68-9010-4c1c9a9ec6ca)

**Comment by telemachus** [comment]
Sidenote, but I think it's easier to read an issue (or discussion) if you copy code in *as code* (i.e., in a code block) rather than pasting images of code.
